pkg/driverbuilder/builder/builders.go

```diff
@@ -116,97 +116,145 @@ func Script(b Builder, c Config, kr kernelrelease.KernelRelease) (string, error)
 }
 
 type GCCVersionRequestor interface {
-	GCCVersion(kr kernelrelease.KernelRelease) float64
+	GCCVersion(kr kernelrelease.KernelRelease) semver.Version
 }
 
 type Image struct {
-	GCCVersion map[kernelrelease.Architecture][]float64
+	GCCVersion map[kernelrelease.Architecture][]string
 }
 
+// We cannot use semver.Version here, because we need to know the
+// actual gcc-$version string that will be used by template scripts,
+// like: make CC=/usr/bin/gcc-{{ .GCCVersion }}
+// When using semver.Version, gcc like semver.Version { Major: 5 }.String() -> "5.0.0"
+// and that would break our scripts, because the real name is just "5"
 var images = map[string]Image{
 	"buster": {
-		GCCVersion: map[kernelrelease.Architecture][]float64{
-			kernelrelease.ArchitectureAmd64: {4.8, 4.9, 5, 6, 8},
-			kernelrelease.ArchitectureArm64: {4.8, 5, 6, 8}, // 4.9 is not present on arm64
+		GCCVersion: map[kernelrelease.Architecture][]string{
+			kernelrelease.ArchitectureAmd64: {"4.8", "4.9", "5", "6", "8"},
+			kernelrelease.ArchitectureArm64: {"4.8", "5", "6", "8"}, // 4.9 is not present on arm64
 		},
 	},
 	"bullseye": {
-		GCCVersion: map[kernelrelease.Architecture][]float64{
-			kernelrelease.ArchitectureAmd64: {9, 10},
-			kernelrelease.ArchitectureArm64: {9, 10},
+		GCCVersion: map[kernelrelease.Architecture][]string{
+			kernelrelease.ArchitectureAmd64: {"9", "10"},
+			kernelrelease.ArchitectureArm64: {"9", "10"},
 		},
 	},
 	"bookworm": {
-		GCCVersion: map[kernelrelease.Architecture][]float64{
-			kernelrelease.ArchitectureAmd64: {11, 12},
-			kernelrelease.ArchitectureArm64: {11, 12},
+		GCCVersion: map[kernelrelease.Architecture][]string{
+			kernelrelease.ArchitectureAmd64: {"11", "12"},
+			kernelrelease.ArchitectureArm64: {"11", "12"},
 		},
 	},
 }
 
-func defaultGCC(kr kernelrelease.KernelRelease) float64 {
-	switch kr.Version {
+func defaultGCC(kr kernelrelease.KernelRelease) semver.Version {
+	switch kr.Major {
 	case 5:
-		if kr.PatchLevel >= 18 {
-			return 12
+		if kr.Minor >= 18 {
+			return semver.Version{Major: 12}
 		}
-		return 11
+		return semver.Version{Major: 11}
 	case 4:
-		return 8
+		return semver.Version{Major: 8}
 	case 3:
-		if kr.PatchLevel >= 18 {
-			return 5
+		if kr.Minor >= 18 {
+			return semver.Version{Major: 5}
 		}
-		return 4.9
+		return semver.Version{Major: 4, Minor: 9}
 	case 2:
-		return 4.8
+		return semver.Version{Major: 4, Minor: 8}
 	default:
-		return 12
+		return semver.Version{Major: 12}
 	}
 }
 
+func mustParseTolerant(gccStr string) semver.Version {
```

Comment:
`ParseTolerant` allows to parse eg: `5` to a `5.0.0` semver. Using just `Parse`, it would panic.